src/blockchain.js

```diff
@@ -35,16 +35,17 @@ class Blockchain{
         }
       }
     }
-    if (!this.checkAgainstTarget(mining_target, block.getHash())) {
+    if (!this.checkAgainstTarget(miningTarget, block.getHash())) {
       return false;
     }
     this.chain.push(block);
     return true;
   }
 
-
   isChainValid(){
     for(let i = 1; i < this.chain.length; i++){
+      if (this.chain.length === 1) {throw new Error("Chain only contains the genesis block")}
```

Comment:
This line should be before the loop, we don't need to enter a for loop if the chains has only genesis block

Reply:
done

Reply:
Will be handled as part of #45